owtf/webapp/src/containers/TargetsPage/TargetsTable.js

```diff
@@ -0,0 +1,448 @@
+    this.setState({
+      data: this.props.targets.slice(currentIndex, currentIndex + sizePerPage),
+      currentPage: page
+    });
+  }
+
+  onSizePerPageList(sizePerPage) {
+    const currentIndex = (this.state.currentPage - 1) * sizePerPage;
+    this.setState({
+      data: this.props.targets.slice(currentIndex, currentIndex + sizePerPage),
+      sizePerPage: sizePerPage
+    });
+  }
+
+  //function responsible for bulk deletion of targets
+  handleDeleteTargets(target_ids) {
+    const target_count = target_ids.length;
+    const status = {
+        complete: [], //contains IDs of targets that are successfully deleted
+        failed: [] //contains IDs of targets that fails to delete
+    };
+    target_ids.map((id) => {
+      this.props.onDeleteTarget(id);
+      //
+      setTimeout(()=> {
+        if(this.props.deleteError !== false){
+          status.failed.push(id);
+        } else{
+          status.complete.push(id);
+        }
+        summarise();
+      }, 200);
+    });
+    const summarise = () => {
+      if(status.complete.length + status.failed.length === target_count){
+        const base_message = status.complete.length + " targets deleted."
+        if (status.failed.length) {
+            this.props.handleAlertMsg("warning", base_message + status.failed.length + " attempts failed.");
+        } else {
+          this.props.handleAlertMsg("success", base_message);
+        }
+      }
+    }
+  }
+
+  // function to delete a single target
+  handleDeleteTarget(target_id) {
+    this.props.onDeleteTarget(target_id);
+    setTimeout(()=> {
+      if(this.props.deleteError !== false){
+        this.props.handleAlertMsg("danger", "Server replied: " + this.props.deleteError);        
+      } else{
+        this.props.handleAlertMsg("success", "Target deleted successfully!");
+      }
+      summarise();
+    }, 200);
+  }
+
+  //function responsible for removing all the selected targets at the same time  
+  handleRemoveTargetsFromSession(target_ids) {
+    const target_count = target_ids.length;
+    const status = {
+        complete: [], 
+        failed: []
+    };
+    target_ids.map((id) => {
+      this.props.onRemoveTargetFromSession(this.props.getCurrentSession(), id);
+      setTimeout(()=> {
+        if(this.props.removeError !== false){
+          status.failed.push(id);
+        } else {
+          status.complete.push(id);
+        }
+        summarise();
+      }, 200);
+    });
+    const summarise = () => {
+      if(status.complete.length + status.failed.length === target_count){
+        const base_message = status.complete.length + " targets removed."
+        if (status.failed.length) {
+            this.props.handleAlertMsg("warning", base_message + status.failed.length + " attempts failed.");
+        } else {
+          this.props.handleAlertMsg("success", base_message);
+        }
+      }
+    }
+  }
+
+  // function to delete a single target
+  handleRemoveTargetFromSession(target_id) {
```

Comment:
@sharmamohit123 Don't you think we can utilize `handleRemoveTargetsFromSession` to delete single target too?